compiler/semobjconstr.nim

```diff
@@ -113,6 +113,13 @@ proc branchVals(c: PContext, caseNode: PNode, caseIdx: int,
     for i in 1 .. caseNode.len-2:
       processBranchVals(caseNode[i], excl)
 
+proc rangeTypVals(rangeTyp: PType): IntSet =
+  assert rangeTyp.kind == tyRange
+  let (a, b) = (rangeTyp.n.sons[0].intVal, rangeTyp.n.sons[1].intVal)
+  result = initIntSet()
+  for it in a .. b:
```

Comment:
I'm a bit worried this would produce an enormous set for ``0..1_000_000``. Anything we can do about that? Restrict the feature to small ranges?

Reply:
I started fixing this by adding the same limitation that is used by #11273 (only types with maximum 2^16 possible values allowed). However, I noticed that #11273 only checks the type of the runtime discriminator, not the type of the actual object field. Because of this, this snippet is allowed and takes ~30s to compile on my machine with v0.20.0:
```nim
type Obj = object
  case x: int
  of 0 .. 5_000_000_000.int:
    field: int
  else:
    discard

let x: range[0..10] = 1

case x
of 1:
  let o = Obj(x: x, field: 1)
else: discard
```
Should I fix this as well even though it would be a breaking change? Seems unlikely that changing it would break anything in reality...

Reply:
That's what the "important_packages" are for, so go ahead.